mods/env_sounds/init.lua

```diff
@@ -0,0 +1,97 @@
+-- Parameters
+
+local radius = 8 -- Water node search radius around player
+
+-- End of parameters
+
+
+local handles = {}
+local river_source_sounds = minetest.settings:get_bool("river_source_sounds")
+
+
+-- Update sound for player
+
+local function update_sound(player)
+	local player_name = player:get_player_name()
+	-- Search for water nodes in radius around player
+	local ppos = player:get_pos()
+	local areamin = vector.subtract(ppos, radius)
+	local areamax = vector.add(ppos, radius)
+	local wpos, num
+	if river_source_sounds then
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_source",
+				"default:river_water_flowing"
+			}
+		)
+	else
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_flowing"
+			}
+		)
+	end
+	-- Total number of waters in radius
+	local waters = #wpos
+	if waters == 0 then
+		return
+	end
+
+	-- Find average position of water positions
+	local wposav = vector.new()
+	for i, pos in ipairs(wpos) do
+		wposav = vector.add(wposav, pos)
+	end
+	wposav = vector.divide(wposav, waters)
+	-- Play sound
+	local handle = minetest.sound_play(
+		"env_sounds_water",
+		{
+			pos = wposav,
+			to_player = player_name,
+			gain = math.min(0.04 + waters * 0.004, 0.4),
+			max_hear_distance = 32,
+		}
+	)
+	-- Store sound handle for this player
+	if handle then
+		handles[player_name] = handle
+	end
```

Comment:
What happens to the old handle?
Is there even a reason for the `handles` table if it's only used to stop sounds for leaving players (which causes nothing)?

Reply:
Indeed it seems the table isn't needed, i don't know why i 'stop'ped the sound for leaving players, as that seems unnecessary.
Will test with table removed and alter if fine.

Reply:
Tested, it works without the handles table as expected. Removed.
I think this code was a left over from a previous version where i needed the handles.